### PR TITLE
splittable RequestStream

### PR DESCRIPTION
**Pull request by yu-re-ka**
#83

### CODE REVIEW

File: h3/src/quic.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ impl<'a, E: Error + 'a> From<E> for Box<dyn Error + 'a> {
 /// Trait representing a QUIC connection.
 pub trait Connection<B: Buf> {
     /// The type produced by `poll_accept_bidi()`
-    type BidiStream: SendStream<B> + RecvStream;
+    type BidiStream: BidiStream<B>;
```
Comment (seanmonstar) [inline h3/src/quic.rs:35]: If we keep the previous bounds, then implementors aren't _required_ to provide splittable streams, just send and receive. If they implement BidiStream, as an opt-in, then the streams can be splittable.

But by not requiring it, we allow QUIC implementations that don't have an internal cheap way to split.

**Comment by yu-re-ka** [comment]
I redesigned the split mechanism with that use case in mind, please have another look

File: h3/src/frame.rs [hunk 2]
```diff
@@ -12,26 +13,21 @@ use crate::{
...
-    S: RecvStream,
-{
+pub struct FrameStream<S, B> {
     stream: S,
     bufs: BufList<Bytes>,
     decoder: FrameDecoder,
     remaining_data: usize,
     /// Set to true when `stream` reaches the end.
     is_eos: bool,
+    _phantom_buffer: PhantomData<B>,
```
Comment (seanmonstar) [inline h3/src/frame.rs:27]: Was this marker required? How come? I imagine probably yes, but I'm struggling to catch exactly where/why...

Comment (yu-re-ka) [reply h3/src/frame.rs:27]:
```
impl<S, B> FrameStream<S, B>
where
    S: BidiStream<B>,
    B: Buf,
{
```

If FrameStream doesn't have the B parameter, this gives an error that B is unconstrained...

Comment (yu-re-ka) [reply h3/src/frame.rs:27]: #94 would solve this

**Review by seanmonstar — APPROVED** [review]

**Pull request merged**